react/week3/todo-app/src/todolist.js

```diff
@@ -0,0 +1,171 @@
+import { useState, useEffect } from "react";
+import PropTypes from 'prop-types';
+import Timer from "./timer";
+
+const API_URL =
+  "https://gist.githubusercontent.com/benna100/391eee7a119b50bd2c5960ab51622532/raw";
+
+function TodoItem({ id, description, deadline, onDelete, onUpdate }) {
+  const [isChecked, setIsChecked] = useState(false);
+  const [isEdit, setIsEdit] = useState(false);
+  const [updatedDescription, setUpdatedDescription] = useState(description);
+
+  const check = () => {
+    setIsChecked(!isChecked);
+  };
+
+  return (
+    <li className={isChecked ? "checked" : "not-ckecked"}>
+      { isEdit
+      ? <input
+            type="text"
+            value={updatedDescription}
+            onChange={(event)=> {
+                const value = event.target.value;
+                setUpdatedDescription(value);
+            }}/>
+      : <span>#{id} | {description} | {deadline}</span> }
+
+      <input type="checkbox" onChange={check} />
```

Comment:
You might even want to set the value based on the state:
`<input type="checkbox" onChange={check} checked={isChecked} />`